<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>590</width>
    <height>429</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_10">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QTreeWidget" name="treeWidget">
        <column>
         <property name="text">
          <string>key</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>value</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QPushButton" name="deleteFromTreeButton">
            <property name="text">
             <string>Delete from Tree</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="balanceButton">
            <property name="text">
             <string>Balance Tree</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="insertInTreeButton">
            <property name="text">
             <string>Insert in Tree</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QLabel" name="searchLabel">
            <property name="text">
             <string>input key</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLineEdit" name="searchLineEdit"/>
            </item>
            <item>
             <widget class="QPushButton" name="searchButton">
              <property name="text">
               <string>Search in Tree</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QPushButton" name="traversInorderButton">
          <property name="text">
           <string>TraversInorder</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="traversPosorderButton">
          <property name="text">
           <string>TraversPosorder</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="traversPreorderButton">
          <property name="text">
           <string>TraversPreorder</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="numberLabel">
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="numberButton">
          <property name="text">
           <string>number of nods in left</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QTreeWidget" name="arrayTreeWidget">
          <column>
           <property name="text">
            <string>key</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>value</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="FromArrayToTreeButton">
          <property name="text">
           <string>From array to Tree</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>key</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>value</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLineEdit" name="keyLineEdit"/>
            </item>
            <item>
             <widget class="QLineEdit" name="valueLineEdit"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="addToArrayButton">
          <property name="text">
           <string>Add to array</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
